deepfence_bootstrapper/controls/controls.go

```diff
@@ -181,17 +169,18 @@ func SetAgentControls() {
 	err = router.RegisterControl(ctl.StopComplianceScan,
 		func(req ctl.StopComplianceScanRequest) error {
 			log.Info().Msg("StopComplianceScanRequest called")
-			scanId, ok := req.BinArgs["scan_id"]
-			var err error
-			if ok {
-				retVal := linuxScanner.StopScan(scanId)
-				if !retVal {
-					err = errors.New("Failed to stop scan")
-				}
-			} else {
-				err = errors.New("Missing scan id in the StopComplianceScanRequest")
+			scanID, ok := req.BinArgs["scan_id"]
+
+			if !ok {
+				return errors.New("missing scan id in the StopComplianceScanRequest")
+			}
+
+			// TODO(tjonak): random boolean return value. Should be enum instead.
+			if retVal := linuxScanner.StopScan(scanID); !retVal {
+				return errors.New("failed to stop scan")
 			}
```

Comment:
Ideally this should return an error by itself - easy to change?

Reply:
Yup, due to amount of changes I didn't want to touch actual logic unless too much. Gonna adjust.